docs/doc-contributing-documents.md

```diff
@@ -0,0 +1,389 @@
+
+In order to guide usage and disambiguate synonyms, we characterise each document type along three dimensions of the commitment that is being taken in it:
+
+1. The `writer` of the document;
+2. the targeted `audience`;
+3. the `object` of the commitment.
+
+A document type thus looks like:
+
+```json
+{
+  …
+  "Privacy Policy": {
+    "commitment": {
+      "writer": "service provider",
+      "audience": "end user",
+      "object": "personal data"
+    }
+  },
+  …
+}
+```
+
+Please note that we do not want [service-specific types](https://github.com/ambanum/OpenTermsArchive/pull/89) such as “Twitter Privacy Policy”. Document types should be generic, even if only one service uses them at a given time. Otherwise, duplication occurs and [important efforts](https://github.com/ambanum/OpenTermsArchive/pull/88) have to be deployed to deduplicate. The triptych form “writer / audience / object” is precisely used to avoid this sort of duplication.
+
+### Testing your declaration
+
+You can test the declarations you created or changed by running the following command:
+
+```
+npm test [$service_id [$another_service_id …]]
+```
+
+Since this operation fetches documents and could be long, you can also validate the declaration structure only:
+
+```
+npm run test:schema [$service_id [$another_service_id …]]
+```
+
+#### Linting
+
+In order to ensure consistency across declarations, all declarations files have to be formatted homogeneously.
+
+In order to achieve this, you can use the following command:
+
+```
+npm run lint [$service_id [$another_service_id …]]
+```
+
+## Maintaining declarations
+
+All parts of a **document** **declaration** (web location, selection, noise removal, single or multiple pages distribution…) can change over time. The process of updating these elements to enable continued **tracking** is called **maintenance**. Without it, **documents** can become:
+
+- **unreachable**: no **snapshot** can be **recorded** at all, because the **location** changed or the **service** denies access;
+- **unextractable**: no **version** can be **extracted** from the **snapshot**, because the selection of content or some **filter** fails;
+- **noisy**: both **snapshots** and **versions** are **recorded** but the **changes** contain **noise** that should have been **filtered out**.
+
+Open Terms Archive needs to keep track of this changes in order to regenerate versions history from snapshots history.
+
+### Service history
+
+To keep track of services declarations and filters changes, Open Terms Archive offers a versioning system. It is optional and should be added only when needed. It works by creating history files for documents declarations and filters, where each entry should be a previous valid declaration or filter function and should have an expiry date.
```

Comment:
It seems to me that it should be made clear why this is important to keep track of services delcarations and filters changes.

Reply:
This is supposed to be done exactly in the paragraph before the one you quoted, and apparently we failed. Could you please clarify what information you're missing? 🙂

Reply:
As seen synchronously, @clementbiron withdraws that comment.

Reply:
ok I reread it and it seems clear to me now.